File: apps/server-car-rental/src/APIs/Car/Dtos/CarWhereInput.cs
namespace CarRentalService.APIs.Dtos;

public class CarWhereInput
{
    public DateTime CreatedAt { get; set; }
    public ICollection<RentalDto>? Rentals { get; set; }
    public decimal? RentalPricePerDay { get; set; }
    public string? RegistrationNumber { get; set; }
    public string? Make { get; set; }
    public string? Model { get; set; }
}

File: apps/admin-car-rental/src/api/car/CarWhereInput.ts
import { StringFilter } from "../../util/StringFilter";
import { RentalListRelationFilter } from "../rental/RentalListRelationFilter";
import { FloatNullableFilter } from "../../util/FloatNullableFilter";
import { StringNullableFilter } from "../../util/StringNullableFilter";
import { OrderListRelationFilter } from "../order/OrderListRelationFilter";

export type CarWhereInput = {
  id?: StringFilter;
  rentals?: RentalListRelationFilter;
  rentalPricePerDay?: FloatNullableFilter;
  status?: "Option1";
  registrationNumber?: StringNullableFilter;
  make?: StringNullableFilter;
  model?: StringNullableFilter;
  orders?: OrderListRelationFilter;
};
